### ISSUE TITLE
"Fatal error Docker is already running" after 2.3.0.2 upgrade

### ISSUE THREAD

**Issue by wpwood** [docker/for-mac#4556]
- [X] I have tried with the latest version of my channel (Stable or Edge)
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior

Docker runs without error after rebooting my system

### Actual behavior

After upgrading from Docker Desktop Community 2.2.0.5 to Docker Desktop Community 2.3.0.2, Docker starts successfully. However, I have the `Start Docker Desktop when you log in` option selected, and after I reboot my system and log in, I get the following error modal:

<img width="618" alt="Screen Shot 2020-05-13 at 8 24 59 AM" src="https://user-images.githubusercontent.com/1039525/81811955-6f178e80-94f3-11ea-9487-4faf94160ce7.png">

### Information

  - macOS Version: macOS Catalina 10.15.4

This problem is new with 2.3.0.2. It did not happen with 2.2.0.5. It happens every time I reboot and log in.

If I then select `Exit` from the `Fatal error` modal and start Docker by clicking on the icon in the Applications folder, it starts fine.

Thinking that there might be some bad, leftover data from my 2.2.0.5 install, I completely uninstalled Docker, then reinstalled 2.3.0.2, and I see the same issue.

One thing of note: My company uses a VPN that is started automatically on login, but it's possible that Docker Desktop is trying to start before the VPN. Is it possible that this is interfering? In any case, it worked fine with the previous Docker Desktop version.

### Diagnostic logs

```
Docker for Mac: version...

```

### Steps to reproduce the behavior

  1. Install Docker Desktop Community 2.3.0.2
  2. Reboot and log in

**Comment by wpwood** [comment]
Additional note: I turned off the VPN on my system and rebooted. The VPN didn't start, but I still received the error. So, it's not interference from the VPN.

**Comment by alex-scown-checkit** [comment]
Using the Dashboard option in the Docker menu works - but it feels odd that opening the app icon is tied to the Docker daemon and not the dashboard app (particularly as the app doesn't appear to be running in the Dock).

This is on macOS 10.15.4, and did not occur prior to 2.3.0.2.

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/33029283/81962334-7c27a100-960b-11ea-8036-9d4d542decd1.png">

**Comment by wpwood** [comment]
I was able to resolve this. It turns out that, I assume from previous installs, I had several `Docker` entries, as well as a `Docker Desktop` entry in my `Login Items` for my User. I removed them all, then selected the `Start Docker Desktop when you log in`, which added one back.

I now have one, and when I reboot, I no longer get that error and Docker comes up.

<img width="679" alt="Screen Shot 2020-05-28 at 5 14 58 PM" src="https://user-images.githubusercontent.com/1039525/83194613-e90c6200-a106-11ea-8798-eae8bb4725b5.png">
